src/xtgeo/xyz/_xyz.py

```diff
@@ -413,35 +404,27 @@ def to_roxar(
     # Get and Set properties
     # ==================================================================================
 
-    # @abc.abstractproperty
-    # def nrow(self):
-    #     """NROW"""
-
     @property
-    @abc.abstractmethod
     def dataframe(self):
-        """Dataframe"""
+        """Returns or set the Pandas dataframe object."""
+        return self._df
 
     @dataframe.setter
-    @abc.abstractmethod
     def dataframe(self, df):
-        """Dataframe setter"""
-
-    # @abc.abstractmethod
-    # def get_carray(self, lname):
-    #     """Returns the C array pointer (via SWIG) for a given log.
-
-    #     Type conversion is double if float64, int32 if DISC log.
-    #     Returns None of log does not exist.
-    #     """
-    #     try:
-    #         np_array = self._df[lname].values
-    #     except Exception:
-    #         return None
-
-    #     if self.get_logtype(lname) == 'DISC':
-    #         carr = _xyz_lowlevel.convert_np_carr_int(self, np_array)
-    #     else:
-    #         carr = _xyz_lowlevel.convert_np_carr_double(self, np_array)
-
-    #     return carr
+        self._df = df.copy()
+
+    @property
+    def nrow(self):
+        """Returns the Pandas dataframe object number of rows."""
+        if self.dataframe is None:
+            return 0
+        return len(self.dataframe.index)
+
+
+def inherit_docstring(inherit_from):
```

Comment:
It's fairly general this function, could perhaps be placed somewhere else. And it's the classes in the other files inheriting from the class in this file that would actually use it..

Reply:
I moved it to `common`, not sure it is the best place, but should be better?